src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -249,12 +344,21 @@ void ov::npuw::LLMInferRequest::infer_prefill(ov::SoPtr<ov::ITensor> input_ids,
     pad_position_ids(padded_position_ids, position_ids);
 
     m_prefill_request->infer();
+    kvcache_desc.num_stored_tokens += static_cast<uint32_t>(input_ids->get_shape()[INPUT_IDS_SEQ_LEN_DIM]);
+
+    if (m_tail_mm_request) {
+        LOG_DEBUG("Calling inference for tail model.");
+        m_tail_mm_request->set_tensor(
+            m_tail_embed_port,
+            m_prefill_request->get_tensor(m_prefill_out_ports.at(ov::npuw::LLMCompiledModel::output_embeds)));
```

Comment:
Instead setting this every time you could set the prefill & generate requests' output tensors to the LM head's input one. In that case, depending on which stage is running it always writes to the head's input tensor directly.

In other words, instead of setting the tail's request input tensor every time you could set it once to prefill&generate outputs.